bin/metrics-curl.rb

```diff
@@ -58,10 +57,15 @@ class CurlMetrics < Sensu::Plugin::Metric::CLI::Graphite
          default: "#{Socket.gethostname}.curl_timings"
 
   def run
+    `type -t "curl" > /dev/null 2>&1`
```

Comment:
I think it would be better to use native methods over shelling out: http://ruby-doc.org/stdlib-2.4.1/libdoc/mkmf/rdoc/MakeMakefile.html#method-i-find_executable

Reply:
I'll refactor

Reply:
Ah now i remember. I tried find_executable first.. but it prints to stdout and there's doesn't seem to be a way to force it to run silently.

Even this doesn't work:

    orig_stderr = $stderr.clone
    orig_stdout = $stdout.clone
    $stderr.reopen File.new('/dev/null', 'w')
    $stdout.reopen File.new('/dev/null', 'w')
    find_executable("curl")
    $stdout.reopen orig_stdout
    $stderr.reopen orig_stderr

This outputs  "no" or "yes to stdout  even though i've forcible redirect stdout to /dev/null
It's crazy!

Reply:
thats really odd, I am not by a computer right now but I am honestly curious about this so I will check it out later but ya I guess the shell out makes sense then.